src/migrations.js

```diff
@@ -24,7 +21,7 @@ require.scopes.migrations = (function() {
 var exports = {};
 exports.Migrations= {
   changePrivacySettings: function() {
-    if (!chrome.extension.inIncognitoContext) {
+    if (!chrome.extension.inIncognitoContext && chrome.privacy ) {
```

Comment:
So the specific context where this is important for private browsing is here: https://github.com/EFForg/privacybadgerchrome/blob/master/src/storage.js#L406-L411
The migrations code is only neccessary as a guard clause around a migration that doesn't run in firefox anyway so we don't need to worry about it. 
I'm not sure that marking the tab as incognito matters in this case, because the storage engine doesn't know or care what tab it's storing data for.

But if split mode doesn't wokr that's a big issue and maybe a blocker for firefox. 
The other issue that I was seeing here was that the tabData structure doesn't seem to work in a private browsing window, somehow the tab ids that it has are different from teh tab ids that the private window thinks it has. This is probably the cause of the popup issue that you saw.